src/audio/eq_fir.c

```diff
@@ -547,29 +556,80 @@ static int eq_fir_prepare(struct comp_dev *dev)
 		return ret;
 
 	/* Initialize EQ */
-	cd->eq_fir_func = eq_fir_passthrough;
 	if (cd->config) {
 		ret = eq_fir_setup(cd, dev->params.channels);
 		if (ret < 0) {
 			comp_set_state(dev, COMP_TRIGGER_RESET);
 			return ret;
 		}
+		switch (config->frame_fmt) {
+		case SOF_IPC_FRAME_S16_LE:
+			trace_eq("f16");
+#if FIR_GENERIC
```

Comment:
Instead of having all this condition code, can we have a block of conditional code at the start of the file that creates some static inline function to set this mapping for HIFI3 and generic modes. That way we only have one area that is conditionally compiled (and makes it easier to add HIFI2 or HIFI4)